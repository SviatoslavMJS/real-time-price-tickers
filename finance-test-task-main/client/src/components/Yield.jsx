import s from '../styles/Yield.module.css';

function Yield(props) {
    return (
        <div className={false ? s.yieldGreen : s.yieldRed}>
            {'+ ' + props.yield}
        </div>
    )
}

export default Yield;