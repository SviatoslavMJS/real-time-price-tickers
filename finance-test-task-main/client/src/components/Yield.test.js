import renderer from 'react-test-renderer';
import Yield from './Yield.jsx';

test('On change prop vector in Yield class must change from green to red / text from +1.34 to -1.34', () => {
    const component = renderer.create(
      <Yield yield={"1.34"} vector={true}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.vector = false;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.vector = true;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
});

test('On change prop yield in Yield text must change from 222 to 555', () => {
  const component = renderer.create(
    <Yield yield={"222"} vector={true}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.yield = "555";

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.yield = "222";

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
});