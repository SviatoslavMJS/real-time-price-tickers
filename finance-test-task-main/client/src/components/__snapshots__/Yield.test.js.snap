// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On change prop vector in Yield class must change from green to red / text from +1.34 to -1.34 1`] = `
<div
  className="yieldGreen"
>
  + 1.34
</div>
`;

exports[`On change prop vector in Yield class must change from green to red / text from +1.34 to -1.34 2`] = `
<div
  className="yieldGreen"
>
  + 1.34
</div>
`;

exports[`On change prop vector in Yield class must change from green to red / text from +1.34 to -1.34 3`] = `
<div
  className="yieldGreen"
>
  + 1.34
</div>
`;

exports[`On change prop yield in Yield text must change from 222 to 555 1`] = `
<div
  className="yieldGreen"
>
  + 222
</div>
`;

exports[`On change prop yield in Yield text must change from 222 to 555 2`] = `
<div
  className="yieldGreen"
>
  + 222
</div>
`;

exports[`On change prop yield in Yield text must change from 222 to 555 3`] = `
<div
  className="yieldGreen"
>
  + 222
</div>
`;
